source/slang/slang-ir-diff-jvp.cpp

```diff
@@ -392,7 +487,7 @@ struct JVPTranscriber
     // in the differential function. We detect and avoid replicating 
     // side-effect instructions.
     // 
-    bool requiresPrimalClone(IRBuilder*, IRInst* instP)
+    bool hasNoSideEffects(IRBuilder*, IRInst* instP)
```

Comment:
there is already a `hasSideEffect` function defined in slang-ir.h

Reply:
Hmm.. I don't think it has the same definition of side-effect that I'm using though. For one, stores into local variables are ok, and certain calls too. Maybe I should rename this to `isPurelyFunctional()` ?

Reply:
I see, that seems a better name. Normally you shouldn't be seeing local variable stores after SSA pass.